docs/customization/api.rst

```diff
@@ -96,3 +106,346 @@ The next step is to modify the security configuration in ``config/packages/secur
+How to customize serialization?
+-------------------------------
+
+Let's say that you want to change the serialized fields in your responses.
+For an example we will use ``Product`` resource and customize its fields.
+
+Adding a field to response
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Let's say that you want to serialize existing field named ``averageRating`` to ``Product`` in admin response so the administrator would be able to check what is products average rating.
+
+First let's copy serialization configuration file named ``Product.xml`` from ``%kernel.project_dir%/vendor/sylius/sylius/src/Sylius/Bundle/ApiBundle/Resources/config/serialization/``
+to ``config/serialization/Product.xml``
+
+Then let's find the attribute ``averageRating``:
+
+.. code-block:: xml
+
+    <!--...-->
+    <attribute name="averageRating">
+            <group>shop:product:read</group>
+    </attribute>
+    <!--...-->
+
+
+and add serialization group that is used by endpoint we want to modify
+
+.. tip::
+
+    You can create your own serialization group for every endpoint or use the one out of the box. If you don't know the name of group for endpoint you want to modify,
+    you can find it by searching for your class configuration file in `%kernel.project_dir%/vendor/sylius/sylius/src/Sylius/Bundle/ApiBundle/Resources/config/api_resources``
+    and look for path that you want to modify.
+
+In this case the new ``group`` is called ``admin:product:read``:
+
+.. code-block:: xml
+
+    <!--...-->
+    <attribute name="averageRating">
+            <group>admin:product:read</group>
+            <group>shop:product:read</group>
+    </attribute>
+    <!--...-->
+
+After this change your response should be extended with new field:
+
+.. code-block:: javascript
+
+    {
+        //...
+        "id": 123,
+        "code": "product_code",
+        "variants": [
+            "/api/v2/shop/product-variants/product-variant-0",
+        ],
+        "averageRating": 3,
+        //...
+    }
+
+.. tip::
+
+    Read more about API Platform `serialization groups <https://api-platform.com/docs/core/serialization/#using-serialization-groups>`_
+
+
+We were able to add a field that exists in ``Product`` class, but what if you want to extend it with custom fields?
+Let's customize response now with your custom fields serialized in response.
+
+Adding a custom field to response
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Let's say that you want to add a new field named ``additionalText`` to ``Product``.
+First let's create a new serializer that will support our ``Product`` resource:
+
+.. code-block:: php
+
+    <?php
+
+    declare(strict_types=1);
+
+    namespace App\Serializer;
+
+    use Sylius\Component\Core\Model\ProductInterface;
+    use Symfony\Component\Serializer\Normalizer\ContextAwareNormalizerInterface;
+    use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+    use Webmozart\Assert\Assert;
+
+    final class ProductSerializer implements ContextAwareNormalizerInterface
```

Comment:
```suggestion
    final class ProductNormalizer implements ContextAwareNormalizerInterface
```
and adjust this service according to the [API Platform docs](https://api-platform.com/docs/core/serialization/#changing-the-serialization-context-on-a-per-item-basis) and to the [normalizer for Product in Sylius](https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/ApiBundle/Serializer/ProductNormalizer.php)